spring-integration-mongodb/src/test/java/org/springframework/integration/mongodb/config/MongoDbOutboundGatewayXmlTests.java

```diff
@@ -0,0 +1,182 @@
+/*
+ * Copyright 2016 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.integration.mongodb.config;
+
+import static org.junit.Assert.assertEquals;
+
+import java.util.List;
+
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+
+import org.springframework.context.support.ClassPathXmlApplicationContext;
+import org.springframework.data.mongodb.MongoDbFactory;
+import org.springframework.data.mongodb.core.MongoTemplate;
+import org.springframework.integration.endpoint.EventDrivenConsumer;
+import org.springframework.integration.mongodb.rules.MongoDbAvailable;
+import org.springframework.integration.mongodb.rules.MongoDbAvailableTests;
+import org.springframework.integration.support.MessageBuilder;
+import org.springframework.messaging.Message;
+import org.springframework.messaging.PollableChannel;
+
+/**
+ * @author Xavier Padró
+ * @since 5.0
+ */
+public class MongoDbOutboundGatewayXmlTests extends MongoDbAvailableTests {
+	private static final String COLLECTION_NAME = "data";
+
+	@Before
+	public void setUp() throws Exception {
+		MongoDbFactory mongoDbFactory = this.prepareMongoFactory();
+		MongoTemplate mongoTemplate = new MongoTemplate(mongoDbFactory);
+
+		mongoTemplate.save(this.createPerson("Artem"), COLLECTION_NAME);
+		mongoTemplate.save(this.createPerson("Gary"), COLLECTION_NAME);
+		mongoTemplate.save(this.createPerson("Oleg"), COLLECTION_NAME);
+		mongoTemplate.save(this.createPerson("Xavi"), COLLECTION_NAME);
+	}
+
+	@After
+	public void cleanUp() throws Exception {
+		MongoDbFactory mongoDbFactory = this.prepareMongoFactory();
+		MongoTemplate mongoTemplate = new MongoTemplate(mongoDbFactory);
+
+		mongoTemplate.dropCollection(COLLECTION_NAME);
+	}
+
+
+	@Test
+	@MongoDbAvailable
+	public void testSingleQuery() throws Exception {
+		ClassPathXmlApplicationContext context =
+				new ClassPathXmlApplicationContext("outbound-gateway-config.xml", this.getClass());
+
+		EventDrivenConsumer consumer = context.getBean("gatewaySingleQuery", EventDrivenConsumer.class);
+		PollableChannel outChannel = context.getBean("out", PollableChannel.class);
+
+		Message<String> message = MessageBuilder.withPayload("").build();
+		consumer.getHandler().handleMessage(message);
+
+		Message<?> result = outChannel.receive(10000);
```

Comment:
M-m-m. Having such a hard logic I'd say this test class belongs to the `outbound`, not `config`. 
Because here is no any config related tests.